### PR TITLE
feat: One more convenient/common producer transform signature

### PR DESCRIPTION
**Pull request by johanandren**
Pretty common to want to transform envelope to outgoing representation without any async (noticed in sample projects)

### CODE REVIEW

File: akka-projection-grpc/src/main/scala/akka/projection/grpc/producer/scaladsl/EventProducer.scala [hunk 1]
```diff
@@ -152,6 +152,10 @@ object EventProducer {
       registerAsyncMapper[A, B](event => Future.successful(f(event)))
     }
 
+    def registerEnvelopeMapper[A: ClassTag, B](f: EventEnvelope[A] => Option[B]): Transformation = {
```
Comment (patriknw) [inline akka-projection-grpc/src/main/scala/akka/projection/grpc/producer/scaladsl/EventProducer.scala:155]: It's a bit of a challenge to use the same naming between the different transformers, where it makes sense. We also have consumer transformation with `registerPayloadMapper`.

We have registerEnvelopeMapper in the original (released) producer transformation, so we should probably stick to that naming. Meaning that we should rename `registerPayloadMapper`?

**Review by patriknw — COMMENTED** [review]
I agree, useful

File: akka-projection-grpc/src/main/scala/akka/projection/grpc/producer/scaladsl/EventProducer.scala (hunk 1 above)
Comment (johanandren) [reply akka-projection-grpc/src/main/scala/akka/projection/grpc/producer/scaladsl/EventProducer.scala:168]: Aligned names and added some more to make the two transformer APIs looks mostly the same. Added API docs as well.

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**